tests/main/snap-quota-thread/task.yaml

```diff
@@ -0,0 +1,118 @@
+summary: Functional test for thread/task quota-related snap commands.
+
+details: |
+  Functional test for snap thread/task quota group commands ensuring that they are 
+  effective in practice.
+
+# these systems do not support quota groups due to their old systemd versions,
+# we do have a check to do for these systems, but that is just to ensure that
+# we don't allow using quota groups, and that check is done in the snap-quota
+# spread instead
+systems:
+  - -ubuntu-14.04-*
+  - -centos-7-*
+  - -amazon-linux-2-*
+  - -ubuntu-16.04-*
+  - -ubuntu-core-16-*
+
+prepare: |
+  snap install jq
+  snap install test-snapd-stressd --edge --devmode
+  snap install test-snapd-curl --edge --devmode
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  # Spawner spawns 32 workers per default in the snap, but the total
+  # count of threads expected is 33 including the main thread.
+  SPAWNER_MAX_THREADS=33
+
+  echo "Create a group with the spawner snap in it"
+  snap set-quota group-one --thread=16 test-snapd-stressd
```

Comment:
I think this one is `--threads=` now (plural), no?

Reply:
Yes correct, my test were a bit behind on the branch!